[Naren Sreekanth] iP

nareus opened the pull request:
# Duke Pro
> Have an assistant to transform your daily life 
The features are built from these assignments [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html)

**The features include:**
- Task management
- Event planning

**Also take note that:**
1. This is still a work in progress
2. There are more features to come :+1:

Features:

 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)

Here's some code you can use in the `main` method as a Java programmer:
```
public class Main {
    public static void main(String[] args) {
        duke.run();
    }
}
```

Commit:
Add Gradle support

nareus committed:
Level-1

nareus committed:
Level-2

nareus committed:
Level-3

nareus committed:
Level-4

nareus committed:
A-TextUiTesting

nareus committed:
Level 5

nareus committed:
Level-6

nareus committed:
changes to input prompt

nareus committed:
Merge remote-tracking branch 'refs/remotes/origin/master'

nareus committed:
Level-7

nareus committed:
Level-8

nareus committed:
resolve merge conflicts

nareus committed:
add Level-8

nareus committed:
restructure program to OOP format

nareus committed:
Create packages from duke classes

nareus committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

nareus committed:
Set up Gradle to build and run duke

nareus committed:
Add JUnit tests

nareus committed:
Set up Jar file along with fixing bugs for Level 7

nareus committed:
Add JavaDocs to classes, methods and members

nareus committed:
Edit according to coding standards

nareus committed:
Add the find feature

nareus committed:
Merge branch 'branch-A-CodingStandard'

nareus committed:
Resolve merge conflicts

benjytan45678 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,114 @@
+ */
+public class Storage {
+    /** list that stores the tasks */
+    private ArrayList<Task> list = new ArrayList<>();
```
Perhaps it would be better to have a more specific name like taskList?

benjytan45678 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,114 @@
+                String line = s.nextLine();
+                String[] parts = line.split("##");
+                switch(parts[0]) {
+                    case("[E]"):
```
Maybe you could remove the extra indentation in the case statement, as indicated in the Java coding standard

benjytan45678 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,28 @@
+    private boolean isDone = false;
+    private String item;
+
+    public void setDone() {this.isDone = true;}
```
You could probably add a line of space?

benjytan45678 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,36 @@
+    /** Date of the deadline */
+    private LocalDate date;
+
+    public Deadline(String item, LocalDate date) {
```
Maybe you can choose a more specific name for item, such as description?

benjytan45678 reviewed (commented):
Great job! Most of your code was quite concise and simple to understand, except for a few minor naming and coding standard violations as described below.

nareus committed:
Add Checkstyle feature and fix checkstyle issues

nareus committed:
create GUI for duke

samuelcheongws commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,198 @@
+     *  Prints tasks that match the filter
+     * @param filter the string used to filter the tasks
+     */
+    public String printFilteredTaskList(String filter) {
```
Well done, the method names are descriptive.

samuelcheongws commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,198 @@
+            string += list.get(index).getTask() + "\n";
+            storage.writeToFile();
+            return string;
+        } catch (NullPointerException e) {
```
Great catching of exceptions. I was wondering if catching a DukeException might be an alternative!

samuelcheongws commented on src/main/resources/view/MainWindow.fxml:
```diff
@@ -0,0 +1,19 @@
+<?import javafx.scene.layout.AnchorPane?>
+<?import javafx.scene.layout.VBox?>
+
+<AnchorPane maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="-Infinity" prefHeight="600.0" prefWidth="400.0" xmlns="http://javafx.com/javafx/8.0.171" xmlns:fx="http://javafx.com/fxml/1" fx:controller="duke.MainWindow">
```
While this is not a week 3 change, I believe reducing the line length would really enhance readability. A suggestion would be to link the styles to a stylesheet.

samuelcheongws reviewed (commented):
LGTM! Overall I see very consistent effort put to maintain good coding standard. Your code is easy to read. There is no apparent coding standard violations.

nareus committed:
Add assert statements

Lack of assert statements might cause the developer to miss out on
certain flaws present in the program.

Several assert statements have been implemented to prevent this from
happening.

Adding assertions allows the developer to identify obvious flaws i.e
things that the programmer knows to be true about the program or state
of objects and variables.

Assertions is useful to test the code since it helps identify bugs present in the program.

nareus committed:
Edit method length and structuring for code quality

This is done so that the overall code quality of the program is of the
expected level of the module.

Every method is limited to thirty lines and three indentation levels.
The properties of SLAP AND KISS is applied to the program.

Thes rules mentioned above are powerful indicators to check whether the
quality of the code is up to standard.

The guidelines can be found  the CS2103T textbook under code quality.

nareus committed:
Add javadoc comments to new methods in duke class

nareus committed:
Merge pull request #2 from nareus/branch-A-Assertions

Add assert statements

nareus committed:
Merge branch 'master' into branch-A-CodeQuality

nareus committed:
Merge pull request #1 from nareus/branch-A-CodeQuality

Edit methods and classes to improve code quality

nareus committed:
implemet schedule command

nareus committed:
Add screenshot and User Guide

nareus committed:
Edit user guide

nareus committed:
Fix checkstyle errors
